### ISSUE TITLE
Cloudflare provider claims record not found although it exists

### ISSUE THREAD

**Issue by jrake-revelant** [hashicorp/terraform#3002]
Hi, we rely heavily on the cloudflare provider to manage our DNS records for external and internal records in our openstack cloud. however, we seem to have it a strange error after a certain number of entries (> 200 records for one domain) where we get this error after a `terraform plan`:

```
Couldn't find CloudFlare Record ID (353641687) for domain (redacted): Record not found
```

We haven't really found a way to debug it, maybe someone here has a suggestions?

Thanks

**Comment by ziggythehamster** [comment]
This is because of an upstream bug: https://github.com/pearkes/cloudflare/issues/5

Here's the matching Terraform bug: https://github.com/hashicorp/terraform/issues/2551

**Comment by stack72** [comment]
Hi all

The cloudflare provider now uses the official cloudflare API - please can you tell me if this is still an error?

Thanks

Paul

**Comment by craigedmunds** [comment]
This is still an issue in Terraform 0.7.0, I believe it's the same issue. I get:

cloudflare_record.cf_dns: Error finding zone ":domain:": ListZones command failed: error from makeRequest: HTTP status 400: content "{\"success\":false,\"errors\":[{\"code\":6003,\"message\":\"Invalid request headers\",\"error_chain\":[{\"code\":6103,\"message\":\"Invalid format for X-Auth-Key header\"}]}],\"messages\":[],\"result\":null}"

**Comment by craigedmunds** [comment]
Hey @stack72 not sure if you've seen this - worth removing the "waiting-response" label?

**Comment by mitchellh** [comment]
This was fixed within the 0.7.x series when we switched to the official client. I was running into this and fixed it myself. :) If I'm mistaken though and this is a different issue, please open a new issue.
